### ISSUE TITLE
train loss of semantic stage has problem

### ISSUE THREAD

**Issue by missaaoo** [zhvng/open-musiclm#20]
when train the semantic stage, the loss becomes very large at more than 2000 steps，I don't know what is causing this problem， do I need to adjust my training strategy?

**Comment by zhvng** [comment]
maybe try a large batch size and/or lower learning rate?

**Comment by missaaoo** [comment]
> maybe try a large batch size and/or lower learning rate?

yeah, when use lower learning rate, the loss is normal .In addition, I would like to ask, the infer speed of coarse model  is very slow, how to modify the decoding part?

**Comment by zhvng** [comment]
inference code is here: https://github.com/zhvng/open-musiclm/blob/main/open_musiclm/open_musiclm.py#L299-L319. I agree that inference is very slow, and it would be cool to explore ways to speed it up!
